src/common-ssh/ssh.c

```diff
@@ -383,45 +395,23 @@ static int guac_common_ssh_authenticate(guac_common_ssh_session* common_session)
         /* Check if password auth is supported on the server */
         if (strstr(user_authlist, "password") != NULL) {
 
-            /* Attempt password authentication */
-            if (libssh2_userauth_password(session, user->username, user->password)) {
-
-                /* Abort on failure */
-                char* error_message;
-                libssh2_session_last_error(session, &error_message, NULL, 0);
-                guac_client_abort(client,
-                        GUAC_PROTOCOL_STATUS_CLIENT_UNAUTHORIZED,
-                        "Password authentication failed: %s", error_message);
-
-                return 1;
-            }
-
-            /* Password authentication succeeded */
-            return 0;
-
+            return password_auth_with_retry(common_session);
```

Comment:
We'll have to be backwards compatible eventually to push our patches upstream - ok for now. Meaning that we lose specifying the password as an input parameter at the moment.

Reply:
ok